pkg/virt-handler/device-manager/generic_device.go

```diff
@@ -340,3 +345,7 @@ func (dpi *GenericDevicePlugin) healthCheck() error {
 func SocketPath(deviceName string) string {
 	return filepath.Join(pluginapi.DevicePluginPath, fmt.Sprintf("kubevirt-%s.sock", deviceName))
 }
+
+func (dpi *GenericDevicePlugin) Initialized() bool {
+	return dpi.initialized
```

Comment:
Do you need a lock?

Reply:
I didn't think I did? The `Start()` function where the variable is changed is being called from a single go routine and the other location where it is change in  and `Stop()` is called from `Start()` and is ran synchronously. Or am I thinking about this wrong? Since it's in a go routine and being called from code not in the same routine should it have a lock anyway?

Reply:
I think that you are reading the value in the health check. If you change it via `Stop()` in one go-routine you may still read the old value in the health-check.